tsl/test/expected/bgw_job_ddl.out

```diff
@@ -39,53 +44,91 @@ ERROR:  permission denied to reassign objects
 RESET ROLE;
 -- Test that renaming a user changes keeps the job assigned to that user.
 ALTER USER another_user RENAME TO renamed_user;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
-   proc_name    |    owner     
-----------------+--------------
- bgw_test_job_1 | renamed_user
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name  |    owner     
+-------------+------------+--------------
+ public      | more_magic | renamed_user
+(1 row)
+
+-- Test that renaming the procedure also modifies the entry in the
+-- jobs table.
+ALTER PROCEDURE more_magic RENAME TO magic;
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name |    owner     
+-------------+-----------+--------------
+ public      | magic     | renamed_user
+(1 row)
+
+-- Test that modifying the schema also modifies the entry in the jobs
+-- table.
+CREATE SCHEMA frugal;
+ALTER PROCEDURE magic SET SCHEMA frugal;
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name |    owner     
+-------------+-----------+--------------
+ frugal      | magic     | renamed_user
+(1 row)
+
+-- Test that renaming the schema will rename the procedure schema
+START TRANSACTION;
+ALTER SCHEMA frugal RENAME TO wicked;
+SELECT proc_schema, proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+ proc_schema | proc_name |    owner     
+-------------+-----------+--------------
+ wicked      | magic     | renamed_user
 (1 row)
 
+ROLLBACK;
 \set VERBOSITY default
 \set ON_ERROR_STOP 0
+SELECT proc_schema, proc_name, owner
+  FROM _timescaledb_config.bgw_job
+ WHERE id = :job_id;
+ proc_schema | proc_name |    owner     
+-------------+-----------+--------------
+ frugal      | magic     | renamed_user
+(1 row)
+
 -- Test that dropping a user owning a job fails.
 DROP USER renamed_user;
 ERROR:  role "renamed_user" cannot be dropped because some objects depend on it
 DETAIL:  owner of job 1000
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
-   proc_name    |    owner     
-----------------+--------------
- bgw_test_job_1 | renamed_user
-(1 row)
-
+-- Test that dropping the procedure fails since there is a background
+-- job using it.
+DROP PROCEDURE frugal.magic;
```

Comment:
Should add test with DROP PROCEDURE ... CASCADE

Reply:
Done.